jaxrs2/src/main/java/io/opentracing/contrib/jaxrs/server/ServerSpanDecorator.java

```diff
@@ -0,0 +1,103 @@
+package io.opentracing.contrib.jaxrs.server;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.core.MultivaluedMap;
+
+import io.opentracing.Span;
+import io.opentracing.contrib.jaxrs.internal.URIUtils;
+import io.opentracing.tag.Tags;
+
+/**
+ * @author Pavol Loffay
+ */
+public interface ServerSpanDecorator {
+
+    /**
+     * Decorate span by incoming object.
+     *
+     * @param requestContext
+     * @param span
+     */
+    void decorateRequest(ContainerRequestContext requestContext, Span span);
+
+    /**
+     * Decorate spans by outgoing object.
+     *
+     * @param responseContext
+     * @param span
+     */
+    void decorateResponse(ContainerResponseContext responseContext, Span span);
+
+    /**
+     * Adds standard tags: {@link io.opentracing.tag.Tags#SPAN_KIND},
+     * {@link io.opentracing.tag.Tags#HTTP_METHOD}, {@link io.opentracing.tag.Tags#HTTP_URL} and
+     * {@link io.opentracing.tag.Tags#HTTP_STATUS}
+     */
+    ServerSpanDecorator STANDARD_TAGS = new ServerSpanDecorator() {
+        @Override
+        public void decorateRequest(ContainerRequestContext requestContext, Span span) {
+            Tags.SPAN_KIND.set(span, Tags.SPAN_KIND_SERVER);
+            Tags.HTTP_METHOD.set(span, requestContext.getMethod());
+
+            String url = URIUtils.url(requestContext.getUriInfo().getAbsolutePath());
+            if (url != null) {
+                Tags.HTTP_URL.set(span, url);
+            }
+        }
+
+        @Override
+        public void decorateResponse(ContainerResponseContext responseContext, Span span) {
+            Tags.HTTP_STATUS.set(span, responseContext.getStatus());
+        }
+    };
+
+    /**
+     * As operation name provides HTTP path. If there are path parameters used in URL then
+     * spans for the same requests would have different operation names, therefore use carefully.
```

Comment:
I'm almost tempted to not-support this since it's such a common stumbling block for people in practice (i.e., not realizing they're enabling a super-high-cardinality operation name).

Just a thought... I could go either way.